metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/query/ESSearchDAO.java

```diff
@@ -262,7 +269,27 @@ public ScrollResult scroll(@Nonnull List<String> entities, @Nonnull String input
 
     scrollRequestTimer.stop();
     // Step 2: execute the query and extract results, validated against document model as well
-    return executeAndExtract(entitySpecs, searchRequest, postFilters, scrollId, keepAlive, size);
+    return executeAndExtract(entitySpecs, searchRequest, transformedFilters, scrollId, keepAlive, size);
+  }
+
+  /**
+   * Allows filtering on entities which are stored as different indices under the hood by transforming the tag
+   * _entityType to _index and updating the type to the index name.
+   * @param filter The filter to parse and transform if needed
+   * @param indexConvention The index convention used to generate the index name for an entity
+   * @return A filter, with the changes if necessary
+   */
+  static Filter transformFilterForEntities(Filter filter, @Nonnull IndexConvention indexConvention) {
```

Comment:
Why does this need to be static? Without it being static you don't need to pass in the indexConvention method param

Reply:
Doesn't need to be, but it seemed easier to test this way...

Reply:
Got it, yeah you can just new up the ESSearchDAO like is done here: https://github.com/datahub-project/datahub/blob/master/metadata-io/src/test/java/com/linkedin/metadata/ESSampleDataFixture.java#L99

(or convert this to a bean definition and wire it in instead of just the IndexConvention)